Squeezer/src/main/java/uk/org/ngo/squeezer/itemlist/AlarmsActivity.java

```diff
@@ -181,20 +181,20 @@ protected void orderPage(@NonNull ISqueezeService service, int start) {
 
         @Override
         public void onItemsReceived(final int count, final int start, Map<String, String> parameters, final List<AlarmPlaylist> items, Class<AlarmPlaylist> dataType) {
-            if (start == 0) {
-                mAlarmPlaylists.clear();
-            }
+            getUIThreadHandler().post(new Runnable() {
+                @Override
+                public void run() {
+                    if (start == 0) {
+                        mAlarmPlaylists.clear();
+                    }
 
-            mAlarmPlaylists.addAll(items);
-            if (start + items.size() >= count) {
-                getUIThreadHandler().post(new Runnable() {
-                    @Override
-                    public void run() {
+                    mAlarmPlaylists.addAll(items);
+                    if (start + items.size() >= count) {
                         mAlarmView.setAlarmPlaylists(mAlarmPlaylists);
```

Comment:
That seems like a good idea. 

Note that the semantics is slightly different, as the UI is not updated until we have received all the items, which also is how it worked originally.

One drawback is that copy of the list is thrown away when AlarmView is done with it, even more so as all AlarmView does with it, is add a few category items to it.